### ISSUE TITLE
LSP completion and hover breaks when using both `export type` and `export`

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#11461]
Open a TS file and paste this code:

```ts
export type {
  CreateHandle,
  EffectCallback,
  Inputs,
  PropRef,
  Reducer,
  Ref,
  StateUpdater,
} from "https://esm.sh/preact@10.5.14/hooks";
export {
  useCallback,
  useContext,
  useDebugValue,
  useEffect,
  useErrorBoundary,
  useImperativeHandle,
  useLayoutEffect,
  useMemo,
  useReducer,
  useRef,
  useState,
} from "https://esm.sh/preact@10.5.14/hooks";
```

You will see that type completions and hover are not working. Now remove the `export type` statement. Type completions and hover work again.

**Comment by kitsonk** [comment]
Effectively duplicate of #11458 which I am currently working on. There is a logic error in how `import type` and `export type` are handled.

**Comment by kitsonk** [comment]
Yeah, the fix for #11458 works for this:

https://user-images.githubusercontent.com/1282577/126308733-138e0585-6175-4334-81b3-476ae8d080e1.mp4

Closing in favour of #11458

**Comment by lucacasonato** [comment]
Ah! Sorry I missed that issue. Thanks for looking into it :-)

**Comment by kitsonk** [comment]
Well, it was only after I went back through some old vscode_deno issues and put 2 + 2 together with some of my own experiences that I was able to figure out what was going on to look into it.
